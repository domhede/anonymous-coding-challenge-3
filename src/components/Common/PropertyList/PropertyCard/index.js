import React from 'react';

const PropertyCard = ({id, price, agency, mainImage}) => {
  return (
    <div>Property {id}</div>
  );
}

export default PropertyCard;