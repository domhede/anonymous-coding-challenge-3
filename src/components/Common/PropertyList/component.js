import React from 'react';
import styled from 'styled-components';
import PropertyCard from './PropertyCard';

const Header = styled.div`
  text-align: center;
  padding: 8px;
  font-size: 22px;
  margin-bottom: 32px;
`;
const Empty = styled.div`
  text-align: center;
  padding: 64px;
`;
Empty.displayName = 'Empty';

const renderProperties = (properties, listType) => {
  if (!properties.length) {
    return <Empty>You have no saved properties. <br></br>To save, hover over a property and click the add button</Empty>
  }
  return properties.map(property => {
    return <PropertyCard listType={listType} key={property.id} {...property} />
  })
}

const PropertyList = ({listType, properties}) => {
  const headerText = listType === 'all' ? 'Results' : 'Saved Properties';

  return (
    <div>
      <Header>{headerText}</Header>
      <div>
        {renderProperties(properties, listType)}
      </div>
    </div>
  );
}

export default PropertyList;
