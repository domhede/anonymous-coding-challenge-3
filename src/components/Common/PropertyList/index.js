import { connect } from 'react-redux';
import PropertyList from './component';

const mapStateToProps = (state, {listType}) => ({
  properties: state.properties[listType]
});

export default connect(mapStateToProps, null)(PropertyList);