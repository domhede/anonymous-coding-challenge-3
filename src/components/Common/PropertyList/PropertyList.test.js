import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import PropertyList from './component';


describe('PropertyList component', () => {
  it('renders without crashing', () => {
    shallow(
      <PropertyList properties={[]} />
    );
  });
  it('renders correctly when prop: listType=all', () => {
    const rendered = renderer.create(
      <PropertyList properties={[]} listType='all' />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('renders correctly when prop: listType=saved', () => {
    const rendered = renderer.create(
      <PropertyList properties={[]} listType='saved' />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('shows empty state when no properties', () => {
    const wrapper = shallow(
      <PropertyList
        properties={[]} listType='saved'
      />
    );
    expect(wrapper.find('Empty').length).toEqual(1);
  })
});

