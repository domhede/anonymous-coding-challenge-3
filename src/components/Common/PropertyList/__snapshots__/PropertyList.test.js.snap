// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyList component renders correctly when prop: listType=all 1`] = `
<div>
  <div
    className="sc-htoDjs dzeWxo"
  >
    Results
  </div>
  <div>
    <div
      className="sc-dnqmqq cuApdr"
    >
      You have no saved properties. 
      <br />
      To save, hover over a property and click the add button
    </div>
  </div>
</div>
`;

exports[`PropertyList component renders correctly when prop: listType=saved 1`] = `
<div>
  <div
    className="sc-htoDjs dzeWxo"
  >
    Saved Properties
  </div>
  <div>
    <div
      className="sc-dnqmqq cuApdr"
    >
      You have no saved properties. 
      <br />
      To save, hover over a property and click the add button
    </div>
  </div>
</div>
`;
